### PR TITLE
Make more of createParser’s config optional

### PR DESCRIPTION
**Pull request by paulmelnikow**
This is another PR along the lines of #385.

The CLI has a few defaults which the programmatic interface doesn’t. I think it makes sense to harmonize these. In my use case, I’m using the CLi to generate the typings, and a test to check that they’re up to date. It would be better not to have to pass in these arguments (which I wasn’t even aware of when I was using the CLI).

Another way to implement this would be to create a MergedConfig type; happy to go that route if you’d prefer.

Looking forward to your feedback!

### CODE REVIEW

File: src/Config.ts [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
     sortProps?: boolean;
     strictTuples?: boolean;
     skipTypeCheck?: boolean;
     encodeRefs?: boolean;
     extraTags?: string[];
 }
 
-export const DEFAULT_CONFIG: Config = {
-    expose: "export",
+export const DEFAULT_CONFIG = {
```
Comment (domoritz) [inline src/Config.ts:18]: I think this should still be of type `Config`, no?

File: src/Config.ts [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
+    expose: "export" as ExposeOption,
     topRef: true,
-    jsDoc: "extended",
+    jsDoc: "extended" as JSDocOption,
```
Comment (domoritz) [inline src/Config.ts:21]: These annotations may not be necessary with the proposed change above.

File: test/config.test.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ const basePath = "test/config";
 
 function assertSchema(name: string, userConfig: Config & { type: string }, tsconfig?: boolean) {
     return () => {
-        const config: Config = {
+        const config = {
```
Comment (domoritz) [inline test/config.test.ts:16]: Why is this needed? If you don't want all parameters, use `Partial<Config>`.

File: src/Config.ts (hunk 1 above)
Comment (paulmelnikow) [reply src/Config.ts:18]: Yea, it seems like that shoudl work, but unforunately it doesn't.

When it's declared of type Config, TypeScript then believes `{ ...DEFAULT_CONFIG, ... config }` is a Config and so the optionals are still optional. By removing the type annotation, TypeScript correctly infers that the default values being provided here are present, so it know `mergedConfig` has these values present, too.

File: test/config.test.ts (hunk 3 above)
Comment (paulmelnikow) [reply test/config.test.ts:16]: Yes, makes sense 👍

Comment (paulmelnikow) [reply test/config.test.ts:16]: Actually it looks like it can just be `Config`. I'll revert this.

**Comment by paulmelnikow** [comment]
I've pushed an additional commit with the changes requested. Let me know if you'd prefer that I rebase it.

File: src/Config.ts (hunk 1 above)
Comment (domoritz) [reply src/Config.ts:18]: So do we want to say `Required<Config>` instead?

Comment (paulmelnikow) [reply src/Config.ts:18]: I did not know that was a thing! But it doesn't seem to work, because `Required<Config>` doesn't declare `path`, `type`, or `tsconfig`. I tried `Partial<Required<Config>>` too which works here, but has the same problem which is it doesn't know e.g. `expose` is defined on `mergedConfig`.

I could create a second type, though, maybe using `Omit` or one of the related operators.

Comment (domoritz) [reply src/Config.ts:18]: Yeah, let's use `Omit` then. I'd like to avoid `as` as much as possible.

**Comment by domoritz** [comment]
Thank you

**Comment by paulmelnikow** [comment]
Thanks for publishing a great tool!
